Programmers/230918_연속_부분_수열_합의_개수/kwaksj329.js

```diff
@@ -0,0 +1,13 @@
+function solution(elements) {
+  let answer = [];
+  const extended = [...elements, ...elements];
+
+  for (let i = 0; i < elements.length; i++) {
+    for (let j = 0; j < elements.length; j++) {
+      const subset = extended.slice(j, i + j + 1);
+      answer.push(subset.reduce((acc, curr) => acc + curr, 0));
+    }
+  }
```

Comment:
이렇게 하면 `N^3` 시간복잡도가 되는데, `slice` 인덱스 하나만 바꿔서 생각해보면 개선 방법을 찾을 수 있어요!

```
[1, 2, 3] 배열이 있다고 가정

1. slice(j, i + j + 1) (현재)
i = 0, j = 0일 때:
subset = [1]

i = 0, j = 1일 때:
subset = [2]

i = 0, j = 2일 때:
subset = [3]

2. slice(i, i + j + 1)
i = 0, j = 0일 때:
subset = [1]

i = 0, j = 1일 때:
subset = [1, 2]

i = 0, j = 2일 때:
subset = [1, 2, 3]
```
2번처럼 `slice`의 첫번째 인자를 `i`로 바꿔준다면 배열 안의 값이 하나씩 늘어나는 것을 볼 수 있죠. 배열을 이렇게 자르게 된다면 요소가 하나 늘어날 때마다 이전 `sum`에 새로운 요소의 값을 더해 `push`해주기만 하면 `N^2`로 해결이 가능해요!
```suggestion
  for (let i = 0; i < elements.length; i++) {
    let sum = 0;
    for (let j = 0; j < elements.length; j++) {
      sum += extended[i + j];
      answer.push(sum);
    }
  }
```